vllm/v1/attention/backends/pallas.py

```diff
@@ -86,6 +86,11 @@ def get_max_num_seqs(model_len: int, page_size: int) -> int:
     # spill less likely. Meanwhile we make sure the page size is in [16, 256].
     @staticmethod
     def get_page_size(vllm_config: VllmConfig) -> int:
+        # TODO: This is a temporary fix for vmem OOM.
+        # For long model length, we use 16 pages to avoid too much VMEM spill.
+        # A more robust solution should be implemented to handle VREG spills.
+        if vllm_config.model_config.max_model_len > 8192:
+            return 16
```

Comment:
Is it a good idea to limit the number of q_token_per_block and  (page_size * page_num_per_blk) instead of reducing page size globally? 

Because I think register spills is directly impacted by kernel block size.

cc @bythew3i